Integración de Cleverbot (#8)

jvicu2001 opened the pull request:
modo de uso
`@bot <mensaje>`

requiere una API_KEY de cleverbot, la cual se puede conseguir en https://www.cleverbot.com/api/

jvicu2001 committed:
Añadido !cleverbot

jvicu2001 committed:
Añadido !cleverbot

jvicu2001 committed:
Merge remote-tracking branch 'origin/dev' into dev

jvicu2001 committed:
reemplazado !cleverbot por @bot <mensaje>

cumple la misma función pera la forma de ejecutarlo es distinta

jkcgs commented:
Será necesario validar la key? igual se agrega una llamada a la api, y en una de esas el wrapper ya lo hace

jvicu2001 commented:
Si nó se valida intenta procesar el mensaje igual y responde con un 'None'

jkcgs commented:
`if respuesta is None: #la wea invalida xd` ?

jvicu2001 commented:
Devuelve error
`
     Ignoring exception in on_message

    Traceback (most recent call last):
        File "/home/yara/Apps/alexis-bot/lib/python3.6/site-packages/discord/client.py", line 307, in _run_event
            yield from getattr(self, event)(*args, **kwargs)
        File "alexis.py", line 375, in on_message
            await self.send_message(chan, resp)
        File "/home/yara/Apps/alexis-bot/lib/python3.6/site-packages/discord/client.py", line 1152, in send_message
            data = yield from self.http.send_message(channel_id, content, guild_id=guild_id, tts=tts, embed=embed)
        File "/home/yara/Apps/alexis-bot/lib/python3.6/site-packages/discord/http.py", line 198, in request
            raise HTTPException(r, data)

    discord.errors.HTTPException: BAD REQUEST (status code: 400): Cannot send an empty message
`

jkcgs commented:
`discord.errors.HTTPException: BAD REQUEST (status code: 400): Cannot send an empty message`

Si la respuesta es None, simplemente tienes que hacer return o enviar un mensaje de error al chat y después el return. Ese error es layer8 :D

jvicu2001 committed:
Eliminado checkeo inecesario de la API_KEY de CleverBot

en ves de consultar el .json de la página junto a la API_KEY solo se revisa si la API responde con un 'None'

jvicu2001 commented:
Definitivamente fue error layer8. Ahí lo cambié

jvicu2001 committed:
Eliminada dependencia innecesaria

ya no se necesita 'requests' en alexis.py

jkcgs commented:
Buena!

https://github.com/santisteban/alexis-bot/pull/23/files#diff-cbf872cd4989cdf41c7b8e29dc38b482R367
Aquí no estás usando format, el resto se ve bien

jvicu2001 committed:
Correción menor

usando .format()

jvicu2001 committed:
Correción menor

usando .format()

jvicu2001 commented:
Ahora está usando format

jkcgs commented:
Oye pero el format va dentro del paréntesis :thinking: 

`pregunta = text.strip('<@{}>').format(self.user.id)`

jvicu2001 committed:
Merge remote-tracking branch 'origin/dev' into dev

jvicu2001 commented:
Ahora sí xd

jkcgs reviewed (changes requested):
Not bad, pero dentro del formato "<@mencion>" (sin mensaje, la pura mención), haría que el bot responda con "que queri" o "que weá" o algún mensaje random, en vez de que diga, o adicional al "formato: blabla"

santisteban commented on alexis.py:
```diff
@@ -74,6 +75,7 @@ def init(self):
         chan = message.channel
         is_pm = message.server is None
         is_owner = 'owners' in self.config and message.author.id in self.config['owners']
+        cbot = CleverWrap(self.config['cleverbot_key'])
```
La clase no debiera iniciarse en cada mensaje, solo 1 vez.

santisteban commented on alexis.py:
```diff
@@ -356,6 +358,20 @@ def init(self):
+        elif text.startswith('<@{}>'.format(self.user.id)):
+            if text.strip() == '<@{}>'.format(self.user.id):
+                resp = 'Formato: @bot <texto>'
+                await self.send_message(chan, resp)
```
No creo que debiera enviar un mensaje con el formato solo si se menciona, pero sí en caso de que alguien envíe un formato erróneo.

jvicu2001 replied:
No creo que pueda haben un "formato erróneo" si **todo** lo que se menciona despues de la mención al bot es enviado a CleverBot

jvicu2001 committed:
Decir frases aleatoriamente al solamente mencionar al bot

jvicu2001 commented:
Ahí tiene frases

santisteban commented on alexis.py:
```diff
@@ -356,6 +358,20 @@ def init(self):
+                respuesta = cbot.say(pregunta)
+                if respuesta == None:
+                    resp = 'El valor "cleverbot_key" no se ha definido o es inválido.'
+                    await self.send_message(chan, resp)
```
Creo que si ocurrió un error al obtener la respuesta de Cleverbot no debiera mandar un mensaje en el canal que se menciono, más bien dejar una constancia en el logger como warning.

santisteban reviewed (commented):
Pendiente.

jkcgs reviewed (changes requested):
Implementar cambios sugeridos por @santisteban

jvicu2001 committed:
Correcciones respecto a CleverWrap

Ahora solo se especifíca 'cbot' mientras se está iniciando el bot y deja constancia en el logger si se ingresó una API_KEY inválida

jvicu2001 commented:
Implementados los cambios sugeridos por @santisteban

jkcgs commented on alexis.py:
```diff
@@ -51,6 +53,12 @@ def init(self):
 
+        cbotcheck = self.cbot.say('test')
+        if cbotcheck == None:
+            self.log.info('CUIDADO: El valor "cleverbot_key" ("%s") es inválido.',self.config['cleverbot_key'])
```
Usar `self.log.warn` en vez de `self.log.info`

jvicu2001 replied:
ok

santisteban replied:
`self.log.warning` y creo debiera removerse el "CUIDADO:".

jkcgs replied:
Ámbos métodos funcionan :D

jkcgs reviewed (changes requested):
Pendiente

jvicu2001 committed:
usar self.log.warn en vez de self.log.info

santisteban commented on alexis.py:
```diff
@@ -55,7 +55,7 @@ def init(self):
         cbotcheck = self.cbot.say('test')
         if cbotcheck == None:
-            self.log.info('CUIDADO: El valor "cleverbot_key" ("%s") es inválido.',self.config['cleverbot_key'])
+            self.log.warn('El valor "cleverbot_key" ("%s") es inválido.',self.config['cleverbot_key'])
```
`self.log.warning` no `self.log.warn`.

jkcgs commented on alexis.py:
```diff
@@ -356,6 +365,20 @@ def init(self):
+                    pregunta = text.strip('<@{}> '.format(self.user.id))
+                    respuesta = self.cbot.say(pregunta)
+                    if respuesta == None:
+                        await self.send_message(chan ,'No ando con ganas de hablar.')
```
El espacio de la coma va a la derecha, no a la izquierda

jkcgs reviewed (changes requested):
Espacio :D

jvicu2001 committed:
usar self.log.warning en vez de self.log.warn

also cambiar un ' ,' por un ', '

jvicu2001 committed:
espacio :)

jvicu2001 committed:
más espacios :)

jvicu2001 committed:
Cambio de versión

santisteban commented on alexis.py:
```diff
@@ -356,6 +365,20 @@ def init(self):
             await self.send_message(chan, resp)
 
+        #cleverbot (@bot <mensaje>)
+        elif text.startswith('<@{}>'.format(self.user.id)):
```
¿Que pasa si `cbotcheck` fue `None`?

Debiera cambiarse `cbotcheck` a `self.cbotcheck` en el `__init__` y tener un `if self.cbotcheck != None`, así puede iniciarse el bot sin Cleverbot y no tener dramas más adelante.

jvicu2001 replied:
Si 'cbotcheck' es 'None' solo notifica que 'cleverbot_key' es inválido, el bot se inicia de todas maneras

jkcgs replied:
El bot se inicia, pero... que ocurre cuando se llama al "comando"? ¿El bot respondería "No ando con ganas de hablar."? ¿No será mejor simplemente desactivar el "comando"? -> Esto implica agregar `and self.cbotcheck is not None` en la condición de esta línea.

santisteban replied:
Mi intención es que no repita "No ando con ganas de hablar." a cada rato (suponiendo que se inicio sin Cleverbot), ya que los usuarios no sabrían que la parte de Cleverbot esta mala, y entonces que simplemente no responda nada cuando se le menciona.

jvicu2001 replied:
Ok

santisteban replied:
Agregaste lineas demás, como bien dice @jkcgs solo tenias que agregar `and self.cbotcheck != None` al `if`.

jvicu2001 replied:
Son para conservar las frases al mencionar al bot
Si no se quieren conservar le aplico a la tijera

jvicu2001 committed:
Desactivar comando si la API_KEY es inválida

jvicu2001 committed:
menos espacios :)

jvicu2001 committed:
revisar si se usa en PM y sacar las frases si no hay cleverbot
